id.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2014 - Max Ekman <max@looplab.se>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventhorizon
+
+// ID is an unique identifier interface
+type ID interface {
+	String() string
+}
+
+// NewID generates a new id
+// Alternatively for a custom id, any func() ID can be used instead.
+var NewID = func() ID {
```

Comment:
Here is where the factory would be used to set different IDs for each client app.